bazel/bazel_windows.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bazel
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+func (b *bazel) newCommand(command string, args ...string) (*bytes.Buffer, *bytes.Buffer) {
+	b.ctx, b.cancel = context.WithCancel(context.Background())
+
+	args = append([]string{command}, args...)
+	args = append(b.startupArgs, args...)
+
+	if b.writeToStderr || b.writeToStdout {
+		containsColor := false
+		for _, arg := range args {
+			if strings.HasPrefix(arg, "--color") {
+				containsColor = true
+			}
+		}
+		if !containsColor {
+			args = append(args, "--color=yes")
+		}
+	}
+
+	bazelPath := findBazel()
+	b.cmd = exec.CommandContext(b.ctx, bazelPath, args...)
+
+	// windows specific as it assures the bazelPath is always double quoted
+	// which assures we can support paths with whitespaces.
+	// It works by specifying the CmdLine after the exec command has been specified
+	// and by wrapping in double quotes the bazelPath content
+	//
+	// NOTE: SysProcAttr.CmdLine does not exist/is supported to be compiled on other
+	// OS other than Windows which is the reason why newCommand fn was created both
+	// for Windows and Unix
+	bazelPath = fmt.Sprintf("\"%s\"", bazelPath)
```

Comment:
If this is all that's necessary to make it work in windows, make a function like `quote(s string) string { return fmt.Sprintf("%q", bazelPath) }` and call it `bazel.go`. There is no need to break out the entire `newCommand` just to change this one line.

Reply:
Okay, then you can do

```
// in bazel_windows.go
func setProcessAttributes(cmd *exec.Command, bazelPath string, args []string) {
  b.cmd.SysProcAttr = &syscall.SysProcAttr{}
  b.cmd.SysProcAttr.CmdLine = fmt.Sprintf("%s %s", fmt.Sprintf("%q", bazelPath), strings.Join(args[:], " "))
}
```

```
// in bazel_unix.go
func setProcessAttributes(cmd *exec.Command, bazelPath string, args []string) {}
```

I'm very reticent to have large swaths of code duplicated for multiple OSes. This will compile down to be identical in byte code but will be much easier to grok the windows specific stuff.

Reply:
Done